### PR TITLE
Document Camera for Elevator Platforms (Camera_Special7)

### PR DESCRIPTION
**Pull request by engineer124**
This documents the camera used by three vertical platforms in fire and 1 vertical platform in non-mq spirit (entrance shortcut). This is where the camera places itself in two distinct positions, that toggles past a certain player posY threshold. In non-mq fire, these platforms are
1) The platform elevator going from the bottom floor up right after the giant lava room near the entrance
2) The platform elevator right before the room where you get the hammer
3) The platform elevator accessed by calling scarecrows song

This is very hardcoded to the specific 4 uses of this setting.

Also, there's probably a better name than `fovRollParam`.
Also possibly `VERTICAL_PLATFORM` -> `ELEVATOR_PLATFORM`?

### CODE REVIEW

File: src/code/z_camera_data.inc.c [hunk 1]
```diff
@@ -2565,22 +2565,52 @@ OnePointCsFull D_8011D9F4[] = {
     { 0x12, 0xFF, 0x0000, 0x0001, 0x0000, 60.0f, 1.0f, { -1.0f, -1.0f, -1.0f }, { -1.0f, -1.0f, -1.0f } },
 };
 
-Vec3f D_8011DA6C[] = {
-    { 3050.0f, 700.0f, 0.0f }, { 1755.0f, 3415.0f, -380.0f }, { -3120.0f, 3160.0f, 245.0f }, { 0.0f, -10.0f, 240.0f }
+typedef enum {
+    /* 0 */ CAM_VERTICAL_PLATFORM_FIRE_LOWER_FLOOR,
+    /* 1 */ CAM_VERTICAL_PLATFORM_FIRE_EAST_TOWER,
+    /* 2 */ CAM_VERTICAL_PLATFORM_FIRE_WEST_TOWER,
```
Comment (Dragorn421) [inline src/code/z_camera_data.inc.c:2571]: is west/east well-defined in OoT?

**Review by Dragorn421 — COMMENTED** [review]
+1 on `ELEVATOR_PLATFORM`, or maybe `UPDOWN_PLATFORM`

"vertical platform" to me sounds like a vertical surface, like a wall or a cliff, I would use "elevator platforms" or "platform that moves up and down" in sentences

<details>

<summary>
The three fire temple actors using this

</summary>

image

</details>

```
ACTOR_BG_HIDAN_FSLIFT -> CAM_VERTICAL_PLATFORM_FIRE_EAST_TOWER
ACTOR_BG_HIDAN_ROCK type=0 -> CAM_VERTICAL_PLATFORM_FIRE_LOWER_FLOOR
ACTOR_BG_HIDAN_SYOKU -> CAM_VERTICAL_PLATFORM_FIRE_WEST_TOWER
```

File: src/code/z_camera_data.inc.c (hunk 1 above)
Comment (engineer124) [reply src/code/z_camera_data.inc.c:2571]: I named this with the concept of the minimap in mind i.e. "west" is the left side of the minimap (where you climb for the hammer) and "east" is the right side of the minimap (where you play scarecrow for 1 of the 6 gold rupee chests), keeping in mind that's for vanilla dungeons, so I didn't want to name it based on that which wouldn't apply for mq dungeons

Comment (engineer124) [reply src/code/z_camera_data.inc.c:2571]: Noticing that all three vertical elevators are different actors, I could also name it based off the actor. Only reason I defaulted to a spatial-descriptive name was the dependence on hard-coded coordinates, but I'm not committed to it

Comment (Dragorn421) [reply src/code/z_camera_data.inc.c:2571]: Naming it after the actors can only backfire once we Eventually name the actors
These names are fine, I just had the impression/memory OoT was inconsistent about cardinal directions but maybe not? even then, sure

**Review by Dragorn421 — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

File: src/code/z_camera.c [hunk 2]
```diff
@@ -6634,47 +6634,62 @@ s32 Camera_Special5(Camera* camera) {
...
-            // Spirit Temple
-            rwData->index = 3;
-        } else if (playerPosRot->pos.x < 1500.0f) {
-            rwData->index = 2;
-        } else if (playerPosRot->pos.y < 3000.0f) {
-            rwData->index = 0;
+            rwData->index = CAM_ELEVATOR_PLATFORM_SPIRIT_TEMPLE_ENTRANCE;
         } else {
-            rwData->index = 1;
+            // Hardcoded positions in SCENE_HIDAN
```
Comment (fig02) [inline src/code/z_camera.c:6653]: with new scene names merged we can switch this over

Comment (fig02) [reply src/code/z_camera.c:6653]: or just say fire temple like was done in a few places as well

**Review by fig02 — APPROVED** [review]

**Pull request merged**